specs/hotel_spec.rb

```diff
@@ -0,0 +1,363 @@
+      @ritz.make_block(@feb1, @feb5, [1, 2, 3, 4], 0.75)
+      free_rooms = @ritz.find_all_unblocked_rooms(@feb1)
+      free_rooms = free_rooms[0].length.must_equal 16
+
+      free_rooms = @ritz.find_all_unblocked_rooms(@feb1..@feb5)
+      free_rooms[0].length.must_equal 16
+      #
+      @ritz.make_block(@feb1, @feb3, [5, 6], 0.75)
+      free_rooms = @ritz.find_all_unblocked_rooms(@feb1)
+      free_rooms = free_rooms[0].length.must_equal 14
+      free_rooms = @ritz.find_all_unblocked_rooms(@feb1..@feb5)
+      free_rooms[0].length.must_equal 14
+      free_rooms = @ritz.find_all_unblocked_rooms(@feb5)
+      free_rooms = free_rooms[0].length.must_equal 16
+
+      @ritz.make_reservation(@feb1, @feb5)
+      free_rooms = @ritz.find_all_unblocked_rooms(@feb5)
+      free_rooms = free_rooms[0].length.must_equal 16
+    end
+  end
+
+  describe "get_cost" do
+    it "returns the cost of a reservation" do
+      holiday = @ritz.make_reservation(@feb1, @feb5)
+      @ritz.get_cost(holiday).must_equal 1000
+    end
+  end
+
+  describe "unreserved_and_unblocked" do
+    it "returns an array of hashes representing open rooms for a range of dates" do
+      all_available = @ritz.unreserved_and_unblocked(@feb1..@feb5)
+      all_available.must_be_kind_of Array
+      all_available.length.must_equal 5
+      all_available[0].must_be_kind_of Hash
+      all_available[0].length.must_equal 20
+    end
+
+    it "returns an array of hashes representing open rooms for one date" do
+      all_available = @ritz.unreserved_and_unblocked(@feb1)
+      all_available.must_be_kind_of Array
+      all_available.length.must_equal 1
+      all_available[0].must_be_kind_of Hash
+      all_available[0].length.must_equal 20
+    end
+
+    it "if no rooms are available, returns empty array" do
+      20.times do
+        @ritz.make_reservation(@feb1, @feb1)
+      end
+      none_available = @ritz.unreserved_and_unblocked(@feb1)
+      none_available[0].length.must_equal 0
+    end
+
+    it "if you reserve a room, it is not available" do
+      @ritz.make_reservation(@feb1, @feb5)
+      one_reserved = @ritz.unreserved_and_unblocked(@feb1..@feb5)
+      one_reserved[0].length.must_equal 19
+    end
+    it "if you make a block, they are not available" do
+      @ritz.make_block(@feb1, @feb5, [1, 2, 3, 4], 0.75)
+      one_reserved = @ritz.unreserved_and_unblocked(@feb1..@feb5)
+      one_reserved[0].length.must_equal 16
+    end
+
+    it "if you make a block and a reservation, all those rooms are not available" do
+      @ritz.make_block(@feb1, @feb5, [1, 2, 3, 4], 0.75)
+      @ritz.make_reservation(@feb1, @feb3)
+      reservation_and_block = @ritz.unreserved_and_unblocked(@feb1)
+      reservation_and_block[0].length.must_equal 15
+      reservation_and_block = @ritz.unreserved_and_unblocked(@feb5)
+    end
+  end
+
+  describe "find_continious_open_room" do
+    it "must return an hash of room =>prices" do
```

Comment:
Somewhere in these tests, you should have coverage around when a room counts as being  available. My entire list of cases includes:
- Not available:
  - Same dates
  - Overlaps in the front
  - Overlaps in the back
  - Completely contained
  - Completely containing
- Available:
  - Completely before
  - Completely after
  - Ends on the checkin date
  - Starts on the checkout date